service/history/workflow/mutable_state_impl.go

```diff
@@ -1496,6 +1532,16 @@ func (ms *MutableStateImpl) IsWorkflowPendingOnWorkflowTaskBackoff() bool {
 	return false
 }
 
+// GetApproximatePersistedSize returns approximate size of in-memory objects that will be written to
+// persistence + size of buffered events in history builder if they will be force flushed
+func (ms *MutableStateImpl) GetApproximatePersistedSize() int {
+	// include buffered events in the size if they will be flushed
+	if !ms.BufferSizeAcceptable() || !ms.HasStartedWorkflowTask() {
```

Comment:
flush buffer events means they will be removed from mutable state and get flushed to history, so not include them in the mutable state size.